File: frontend/src/services/dashboardService.ts
import api from './api';

export const dashboardService = {
  getStudentDashboard: async () => {
    const response = await api.get('/dashboard/student');
    return response.data;
  },

  getFacultyDashboard: async () => {
    const response = await api.get('/dashboard/faculty');
    return response.data;
  },

  getHodDashboard: async () => {
    const response = await api.get('/dashboard/hod');
    return response.data;
  },

  getAdminDashboard: async () => {
    const response = await api.get('/dashboard/admin');
    return response.data;
  }
};

File: backend/src/main/java/com/acronexus/service/DashboardService.java
package com.acronexus.service;

import com.acronexus.dto.response.AdminDashboardResponse;
import com.acronexus.dto.response.FacultyDashboardResponse;
import com.acronexus.dto.response.HodDashboardResponse;
import com.acronexus.dto.response.StudentDashboardResponse;
import com.acronexus.dto.response.CoordinatorDashboardResponse;

import java.util.UUID;

public interface DashboardService {

    StudentDashboardResponse getStudentDashboard(UUID userId);

    FacultyDashboardResponse getFacultyDashboard(UUID userId);

    CoordinatorDashboardResponse getCoordinatorDashboard(UUID userId);

    HodDashboardResponse getHodDashboard(UUID userId);

    AdminDashboardResponse getAdminDashboard();
}
